nextjs-13/src/components/CreateNewsletterModal/index.tsx

```diff
@@ -0,0 +1,127 @@
+'use client';
+import { Dispatch, SetStateAction, useState } from 'react';
+import Modal from 'react-modal';
+import TextareaAutosize from 'react-textarea-autosize';
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+import ClipLoader from 'react-spinners/ClipLoader';
+
+import requestManager from 'lib/request/manager';
+
+const customStyles = {
+  content: {
+    top: '25%',
+    left: '15%',
+    width: '70%',
+    height: '50%',
+  },
+};
+
+type Props = {
+  modalIsOpen: boolean;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+};
+
+const CreateNewsletterModal = ({ modalIsOpen, setIsOpen }: Props) => {
+  const [name, setName] = useState('');
+  const [content, setContent] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const closeModal = () => {
+    setIsOpen(false);
+    setName('');
+    setContent('');
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    setLoading(true);
+
+    try {
+      await requestManager('POST', 'v1/newsletter', {
+        data: {
+          name,
+          content,
+        },
+      });
+
+      closeModal();
+      setLoading(false);
+      toast.success('Created newsletter success!', {
```

Comment:
For `toast` I think is better to move it to `/lib/toast/makeToast.ts` to make it more reusable without define the styling or position.

And for the `ToastContainer` you can just wrap it in `app.tsx` or `layoujt` as you would need to call toast in other page too.

So after that you just call `makeToast('message', 'error')` without import the `ToastContainer` in every page.

Reply:
I see 👀

Doing that indeed make the code looks a lot better! Thank you for this suggestion, Beckham! 🔥

Resolved [here](https://github.com/nimblehq/js-framework-benchmark/pull/106/commits/c312e7e088f83d969ab2a05cf1e01f8cd58c15af)

Reply:
Actually this suggestion is come from when I did the code challenge with @olivierobert 😅.

We need to thank you him also 😂.